pkgs/development/python-modules/polyline/default.nix

```
@@ -0,0 +1,29 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, six
+}:
+
+buildPythonPackage rec {
+  pname = "polyline";
+  version = "1.4.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0x60lm3ccq9zrcqlzyk041xgr1mi0k9lzyyv3cbbdiq9kb88jzvw";
+  };
+
+  propagatedBuildInputs = [ six ];
+
+  # Tests have dependencies that are pinned to specific versions, which makes
+  # builds that run tests unreliable
+  doCheck = false;
```

the only reason why versions matter is because it tries to do a setuptools test phase, which tries to also do an install of the `tests_require` dependencies. Since the package is using the nose test runner, we can just call it directly with the version available on nixpkgs.

```suggestion
  checkInputs = [ flake8 nose ];
  checkPhase = ''
    nosetests
  '';
```

---

@jonringer, I don't work with Python extensively and haven't used setuptools before. Does setuptools require specific, pinned version numbers (i.e., `==x.x.x`) instead of ranges (i.e., `>=x.x.x`)? I'm not clear on whether the decision to pin the testing packages to particular version numbers is a constraint of the tooling or a subjective choice made by the original developer.

---

no, setuptools doesn't require specific versions, it requires what is specified by `tests_require`. It just happens that this package has pinned version `XX==Y.Y`